ui/src/app/components/runs/filters/multiple-status-filter/multiple-status-filter.component.ts

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright 2018 ABSA Group Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {AfterViewInit, Component, Input, OnDestroy} from '@angular/core';
+import {Subject} from 'rxjs';
+import {ClrDatagridFilterInterface} from '@clr/angular';
+import {DagRunModel} from '../../../../models/dagRuns/dagRun.model';
+import {StatusModel} from '../../../../models/status.model';
+import {ContainsMultipleFilterAttributes} from '../../../../models/search/containsMultipleFilterAttributes.model';
+
+@Component({
+  selector: 'app-multiple-status-filter',
+  templateUrl: './multiple-status-filter.component.html',
+  styleUrls: ['./multiple-status-filter.component.scss']
+})
+export class MultipleStatusFilterComponent implements ClrDatagridFilterInterface<DagRunModel>, AfterViewInit, OnDestroy {
+  @Input() removeFiltersSubject: Subject<any>;
+  @Input() property: string;
+  @Input() statuses: StatusModel[];
+  selectedValues: string[] = [];
+
+  changes = new Subject<any>();
+
+  constructor() {}
+
+  ngAfterViewInit(): void {
+    this.removeFiltersSubject.subscribe(_ => this.onRemoveFilter());
+  }
+
+  toggleStatuses(statusModel) {
+    if (!statusModel.checked) {
```

Comment:
1. If you really need the `checked` boolean, I would not store it directly on the `StatusModel`, but maybe create a map from `StatusModel` to boolean. `checked` only has a meaning within this component, but now spills over e.g. to `dagInstanceStatuses.constants.ts` where it has no meaning.

2. I think you don't need `checked` since it is equivalent with `selectedValues.indexOf(statusModel.name) != -1`. In fact, I think you should remove `checked` to eliminate any risk of inconsistency

Reply:
Used the second suggestion, thanks.